dev-packages/eslint-plugin/rules/localization-check.js

```diff
@@ -0,0 +1,109 @@
+// @ts-check
+/********************************************************************************
+ * Copyright (C) 2021 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+const levenshtein = require('js-levenshtein');
+
+/** @type {import('eslint').Rule.RuleModule} */
+module.exports = {
+    meta: {
+        type: 'problem',
+        fixable: 'code',
+        docs: {
+            description: 'prevent incorrect use of \'nls.localize\'.',
+        },
+    },
+    create(context) {
+        const metadata = require('../data/nls.metadata.json');
+        const messages = new Set(Object.values(metadata.messages)
+            .reduceRight((prev, curr) => prev.concat(curr), [])
+            .map(e => e.replace(/&&/g, '')));
```

Comment:
Could this code be moved out of the `create` function? It looks like that function is run quite frequently (once per file, perhaps?), so it might be better to do this work somewhere it can be globally accessible.